### ISSUE TITLE
MPRIS D-Bus next and previous commands bug when using shuffle

### ISSUE THREAD

**Issue by SheSellsSeaShellsSe** [DeaDBeeF-Player/deadbeef#3120]
### Steps to reproduce the problem
Install 'playerctl' or other MPRIS D-Bus program to control the player.

(additionally one could use a global hotkey manager (such as the one built into the desktop) to call playerctl to make global hotkeys, that control not just DeaDBeeF but many more MPRIS/D-Bus aware softwares if they happen to be running)

### What's going on? Describe the problem in as much detail as possible.

1. Have a playlist of songs.
2. Use "shuffle" mode.
3. Try to use the D-Bus to send a previous/next command  (works 99%)
4. Do that exactly when the player is playing a song which is at beginning/end of playlist. (1% edge case, does not work)

However the buttons in DeaDBeeF's ui work just fine if clicked with the mouse cursor.

So basically there is some kind of an extra check (or lack thereof) which does not take "shuffle" into account if the previous/next comes from D-Bus. It works with the buttons though. It seems to think it is already at the end of a playlist, or at the beginning of a playlist, and does nothing. This is only if the command comes via D-Bus.
